src/betterproto/casing.py

```diff
@@ -136,4 +136,8 @@ def lowercase_first(value: str) -> str:
 
 def sanitize_name(value: str) -> str:
     # https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles
-    return f"{value}_" if keyword.iskeyword(value) else value
+    if keyword.iskeyword(value):
+        return f"{value}_"
+    if not value.isidentifier():
```

Comment:
What's the reasoning here? There are many ways that a string could fail to be a valid identifier, I'm not sure prefixing with an underscore is a treatment well matched to many of them. If your intention is more specific (like just to add a prefix to names that don't start with a letter?) then you should implement that logic.

Reply:
The protobuf compiler only accepts ASCII alphabetical identifiers as field names, so the only inputs to this function that isn't a valid python identifier is an integer prefixed string and so to turn it into a valid python identifier you must prefix it with an underscore.

Reply:
Good to know the problem is more constrained that I thought it might be. Although if your intention here is to add a prefix to any string that starts with a digit, then wouldn't it be better (clearer, less surprising) to explicitly test for the first character being a digit?

Reply:
AFAIK the set of invalid identifiers has no intersection with _ + the invalid identifier, this way I just though was more forwards compatible

Reply:
I think this is not quite a one-way door, and we can maybe add a comment or TODO to refactor later, so we can prioritize a 2.0 release. If nobody disagrees I will resolve this conversation